### PR TITLE
iso_extract: new module to extract files from ISO images

### PR DESCRIPTION
**Pull request by dagwieers**
##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME

iso_extract

##### ANSIBLE VERSION

v2.3

##### SUMMARY

This is the original iso_extract module from the ansible-provisioning project.
We are upstreaming all our modules, and this PR is part of that process.

### CODE REVIEW

**Comment by dagwieers** [comment]
@jhoekx @ribbons Please review !

File: lib/ansible/modules/files/iso_extract.py
Comment (ribbons) [inline lib/ansible/modules/files/iso_extract.py]: Bit of a nitpick & not sure if this is the right time for this, but should this just read 'A list of...' as both comma separated and actual lists in YAML are supported?

**Comment by ribbons** [comment]
Apart from that, looks good to me :+1:.

File: lib/ansible/modules/files/iso_extract.py
Comment (dagwieers) [reply lib/ansible/modules/files/iso_extract.py]: You are right, I fixed it. Thanks !

**Comment by dagwieers** [comment]
@ribbons @jhoekx You probably noticed I kept you both as maintainers. Within the Ansible project this also means you have some responsibilities, but it also grants you some rights.

**Comment by ribbons** [comment]
> @ribbons @jhoekx You probably noticed I kept you both as maintainers. Within the Ansible project this also means you have some responsibilities, but it also grants you some rights.

Sounds good - many thanks.

File: lib/ansible/modules/files/iso_extract.py
Comment (gundalow) [inline lib/ansible/modules/files/iso_extract.py:49]: Anyway to extract a whole directory (and subdirectories) or everything?

File: lib/ansible/modules/files/iso_extract.py
Comment (gundalow) [inline lib/ansible/modules/files/iso_extract.py:118]: Might be worth adding a `note:` to say that only the hash of the file is used for comparisons, not any filemode/permissions/etc.

File: lib/ansible/modules/files/iso_extract.py
Comment (gundalow) [inline lib/ansible/modules/files/iso_extract.py]: Worth mounting readonly?

**Review by gundalow — COMMENTED** [review]
Looks good just a few minor questions that don't block getting this added

File: lib/ansible/modules/files/iso_extract.py
Comment (dagwieers) [reply lib/ansible/modules/files/iso_extract.py:49]: Currently that is not supported (and supporting this is a lot harder than what we currently do).
I will leave it up to the maintainers to whether this functionality is wanted/requested.
(The current use-case is to extract kernel/ramdisk from boot images)

File: lib/ansible/modules/files/iso_extract.py
Comment (dagwieers) [reply lib/ansible/modules/files/iso_extract.py:118]: Yes, good idea. I have added such note.

File: lib/ansible/modules/files/iso_extract.py
Comment (dagwieers) [reply lib/ansible/modules/files/iso_extract.py:49]: BTW I did clarify that files are actual files and not directories.

File: lib/ansible/modules/files/iso_extract.py
Comment (dagwieers) [reply lib/ansible/modules/files/iso_extract.py]: Yes, definitely. Implemented !

**Review by gundalow — APPROVED** [review]

**Comment by dagwieers** [comment]
Time to merge this too :-)

**Pull request merged**
